zebra/zebra_rib.c

```diff
@@ -212,11 +212,29 @@ struct nexthop *route_entry_nexthop_ifindex_add(struct route_entry *re,
 	nexthop->type = NEXTHOP_TYPE_IFINDEX;
 	nexthop->ifindex = ifindex;
 
+	/*
+	* For situation when, real interface will be fake.
+	* For static route we need ifname, because ifindex will be bad
+	*/
+	strncpy(nexthop->ifname, ifindex2ifname (nexthop->ifindex, re->vrf_id), sizeof(nexthop->ifname));
```

Comment:
Thanks for using strncpy here. Can you please break this line so it's under 80 columns per kernel style?